filament/backend/src/metal/MetalState.h

```diff
@@ -256,13 +258,23 @@ using PipelineStateCache = StateCache<MetalPipelineState, id<MTLRenderPipelineSt
 // Depth-stencil State
 
 struct DepthStencilState {
-    MTLCompareFunction compareFunction = MTLCompareFunctionAlways;      // 8 bytes
-    bool depthWriteEnabled = false;                                     // 1 byte
-    char padding[7] = { 0 };                                            // 7 bytes
-
-    bool operator==(const DepthStencilState& rhs) const noexcept {
-        return this->compareFunction == rhs.compareFunction &&
-               this->depthWriteEnabled == rhs.depthWriteEnabled;
+    MTLCompareFunction depthCompare = MTLCompareFunctionAlways;                         // 8 bytes
+    MTLCompareFunction stencilCompare = MTLCompareFunctionAlways;                       // 8 bytes
+    MTLStencilOperation stencilOperationStencilFail = MTLStencilOperationKeep;          // 8 bytes
+    MTLStencilOperation stencilOperationDepthFail = MTLStencilOperationKeep;            // 8 bytes
+    MTLStencilOperation stencilOperationDepthStencilPass = MTLStencilOperationKeep;     // 8 bytes
```

Comment:
wow these are crazy expensive -- 8 bytes per constant!?!
might be worse encoding them with our internal constants and have getters.

Reply:
Yeah :( Metal encodes them as an 8 byte unsigned long, even though there's only a handful of enumeration values. It would probably be safe to simply store them as `uint8_t`s.